pkgs/applications/office/zotero-dev/default.nix

```
@@ -0,0 +1,400 @@
+
+        rm -rf pdf-worker
+        cp -r ${pdf-worker}/lib/node_modules/pdf-worker pdf-worker
+
+        chmod +w . -R
+
+        (
+          cd pdf-reader
+          rm -rf pdf.js
+          cp -Lr ${pdf-reader-pdfjs}/lib/node_modules/pdf.js pdf.js
+        )
+        (
+          cd pdf-worker
+          rm -rf pdf.js
+          cp -Lr ${pdf-worker-pdfjs}/lib/node_modules/pdf.js pdf.js
+        )
+
+        rm -rf translators
+        cp -Lr ${translators}/lib/node_modules/translators-check translators
+
+        rm -rf note-editor
+        cp -Lr ${note-editor}/lib/node_modules/zotero-note-editor note-editor
+
+        chmod +w . -R
+
+        find scripts -type f | xargs sed -i 's/npm ci/#npm ci/g'
+        find scripts -type f | xargs sed -i 's/git/#git/g'
+
+        sed -i 's/npm run build/#npm run build/g' scripts/note-editor.js
+      '';
+
+      installPhase = ''
+        mkdir $out
+        cp -r . $out
+      '';
+    };
+
+  zotero-build = stdenv.mkDerivation {
+    pname = "${pname}-build";
+    inherit version meta;
+    src = fetchFromGitHub {
+      owner = "zotero";
+      repo = "zotero-build";
+      rev = "00e854c6588f329b714250e450f4f7f663aa0222";
+      hash = "sha256-Gvt37jObgSQ10GBYjnCLu5XbUAy3oVTkWPvHbhLF+fw=";
+      fetchSubmodules = true;
+    };
+
+    postPatch = ''
+      sed -i -E "/-aL/a '--chmod=Du=rwx'," xpi/build_xpi
+      #sed -i 's/-aL/-L/' xpi/build_xpi
+    '';
+
+    buildInputs = [ python3 rsync perl ];
+
+    buildPhase = ''
+      python3 xpi/build_xpi -s ${zotero-client}/build -c source -m ${rev}
+    '';
+
+    installPhase = ''
+      mkdir $out
+      cp -r . $out
+    '';
+  };
+
+in
+
+stdenv.mkDerivation {
+  inherit pname version meta;
+
+  src = fetchFromGitHub {
+    owner = "zotero";
+    repo = "zotero-standalone-build";
+    rev = "e9ef6bf21d39cc66f1edefdd5b7429bbaf0c5247";
+    hash = "sha256-NcnbqCN6Pti2KuVX79QLrTk2V/3sMxMrbgRzSTluOtM=";
+    fetchSubmodules = true;
+  };
+
+  patches = [ ./fetchxul.patch ];
+
+  postPatch = ''
+    patchShebangs .
+
+    sed -i 's|LINUX_i686_RUNTIME_PATH=.*|LINUX_i686_RUNTIME_PATH="$DIR/xulrunner/firefox"|' config.sh
+    sed -i 's|LINUX_x86_64_RUNTIME_PATH=.*|LINUX_x86_64_RUNTIME_PATH="$DIR/xulrunner/firefox"|' config.sh
+    sed -i 's|ZOTERO_SOURCE_DIR=.*|ZOTERO_SOURCE_DIR="${zotero-client}"|' config.sh
+    sed -i 's|ZOTERO_BUILD_DIR=.*|ZOTERO_BUILD_DIR="${zotero-build}"|' config.sh
+
+    sed -i -E 's|(.*hash=).*|\1${rev}|' scripts/dir_build
+    sed -i '/build_xpi/d' scripts/dir_build
+
+    sed -i -E 's|(rsync -a.*)|\1; chmod -R +w $BUILD_DIR|' build.sh
+    #sed -i 's|MaxVersion=.*|MaxVersion=111.0|' assets/application.ini
+  '';
+
+  nativeBuildInputs = [ makeWrapper python3 unzip zip perl rsync wrapGAppsHook ];
+  buildInputs = [ gsettings-desktop-schemas glib gtk3 gnome.adwaita-icon-theme dconf ];
+
+  configurePhase = ''
+    mkdir xulrunner
+    cp -Lr ${firefox-esr-102-unwrapped}/lib/firefox xulrunner
```

```suggestion
    cp -Lr ${firefox-esr-unwrapped}/lib/firefox xulrunner
```
or do we need that exact version of firefox? I think the alias is going away with the next esr bump anyway and this would probably be updated.

---

I think we need Firefox 102 precisely, according to this: https://www.zotero.org/support/dev/zotero_7_for_developers. Is there another way to pin the version instead of the package name?

---

Then maybe we should use firefox-esr or add an overwrite for the firefox package but that might break over time, too as options for old firefox's are usually removed.